library/dnet.c

```diff
@@ -1349,15 +1304,15 @@ static int dnet_cmd_bulk_read(struct dnet_backend_io *backend,
 					!dnet_id_cmp_str((const unsigned char *)&ios[i].id, (const unsigned char *)&cmd->id.id);
 		if (use_oplock) {
 			memcpy(&lock_id.id, &ios[i].id, DNET_ID_SIZE);
-			dnet_oplock(backend, &lock_id);
+			dnet_oplock(dnet_backend_get_pool(st->n, read_cmd.backend_id), &lock_id);
 		}
 
-		ret = dnet_process_cmd_raw(backend, st, &read_cmd, &ios[i], 1, cmd_stats->queue_time);
+		ret = dnet_process_cmd_raw(st, &read_cmd, &ios[i], 1, cmd_stats->queue_time);
 		dnet_log(st->n, DNET_LOG_NOTICE, "%s: processing BULK_READ.READ for %d/%d command, err: %d",
 			dnet_dump_id(&cmd->id), (int) i, (int) count, ret);
 
 		if (use_oplock) {
-			dnet_opunlock(backend, &lock_id);
+			dnet_opunlock(dnet_backend_get_pool(st->n, read_cmd.backend_id), &lock_id);
```

Comment:
Can we eliminate double call with the same args of `dnet_backend_get_pool()`?
It might be a good optimization, if compiler won't do so.